### ISSUE TITLE
Request for Changes for Dealing with PRs

### ISSUE THREAD

**Issue by atdservicebot** [cityofaustin/atd-data-tech#7767]
> What application are you using?

Finance & Purchasing

> Describe the problem.

Hi, in the past Adelita Aguilar and myself, Sharon Ballast, were able to make edits to the PRs in various states. I believe I was told we had a dual role set up which was causing extended delays when entering information in the PRs. We were changed to one role and the issues were gone. The change corrected my initial inquiry but what it removed that I want to try to get back in some way is the ability to re-open a PR after it is closed. I was able to delete an item in the past and it re-opened the PR- I can not currently do this. In addition, we would like to be able to Cancel a PR when it is in the Pending Invoice Phase. Please let me know if you need additional information to clarify this request. Thank you!

> Describe the outcome you'd like to see when this feature is implemented.

Ability to re-open a closed PR and the ability to cancel a PR in the Pending Invoice Phase.

> Describe any workarounds you currently have in place or alternative solutions you've considered.

None to re-open. For cancellations- we reduce the item price to .01c and add a fake .01c invoice to close out.

> Is there anything else we should know?

within 30 days if possible.

> Requested By
Error :(

Request ID: DTS21-103040

**Comment by jaime-mckeown** [comment]
@johnclary The PM assignments doc shows you for Finance and Purchasing, is that correct?

**Comment by johnclary** [comment]
@dianamartin you mind looking into this sometime this week?

**Comment by dianamartin** [comment]
I called up Adelita to see if she could walk me through the pain points of their business processes. She said that there are times that when a PR is "Closed" they need to Re-open it to make a correction and they'd like that ability to have that. We had changed their user roles from "User" and "Admin" to just "Admin" and apparently that stopped them from doing that. 

I created a "Re-open" button only for admins, applied page rules to ensure that the Admins were only the users that could see it. When the button was clicked it'd change the `PURCHASE_PROCESSING_COMPLETE` field from "Yes" to "No". 
- [x] Tested that it worked from "Closed" and went to "Processing | Purchasing" status as intended
- [x] Added a new field "Email Requester" (Y/N) only to the "Process Purchase Request" form
  - it defaults to "Yes" so that the email to the users will still work
  - Finance staff will change to "No" when they're making corrections so that the user will not get an email alert
- [x] When the form is completed, it switches the status back to "Closed"

The Finance staff also had issues with when PRs were in "Pending Invoice" status and the requester decides to not pursue it and want to Cancel the PR. The Admins have been doing a work around rather than being able to "Cancel" the PR. 

I adjusted the page rule on the "Cancel Request" page from 
![image](https://user-images.githubusercontent.com/21112975/146960959-c0dff6cb-22d5-45c3-aa0e-ed32859aed8b.png)

to this
![image](https://user-images.githubusercontent.com/21112975/146961037-d908302c-7091-4f9a-b5cb-36eb2020de75.png)
- [x] Tested PR going from "Pending Invoice" to "Cancelled"
